c/tskit/trees.c

```diff
@@ -2558,14 +2738,46 @@ sample_sets_to_bit_array(const tsk_treeseq_t *self, const tsk_size_t *sample_set
     return ret;
 }
 
+static int
+check_sites(const tsk_id_t *sites, tsk_size_t num_sites, tsk_size_t num_site_rows)
+{
+    int ret = 0;
+    tsk_id_t id;
+    tsk_size_t i;
+
+    if (sites == NULL) {
+        goto out;
+    }
+
+    for (i = 0; i < num_sites - 1; i++) {
+        id = sites[i];
```

Comment:
Can use a generic check like this:
```
if (sites[i] < 0 || sites[i] >= num_sites) {
    ret = TSK_ERR_SITE_OUT_OF_BOUNDS
    goto out

}

Reply:
I think this is more straightforward. If we've encountered an out of bounds site, immediately fail.

Since we're checking pairs, we'll still need to handle the last site separately.